ui/components-react/src/components-react/newEditors/values/Values.ts

```diff
@@ -0,0 +1,126 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+/**
+ * Type definition for numeric value that can be handled by editor.
+ * @beta
+ */
+export interface NumericValue {
+  rawValue: number | undefined;
+  displayValue: string;
+  roundingError?: number;
+}
+
+/**
+ * Type definition for instance key value that can be handled by editor.
+ * @beta
+ */
+export interface InstanceKeyValue {
+  key: { id: string; className: string };
+  label: string;
+}
+
+/**
+ * Type definition for text value that can be handled by editor.
+ * @beta
+ */
+export interface TextValue {
+  value: string;
+}
+
+/**
+ * Type definition for boolean value that can be handled by editor.
+ * @beta
+ */
+export interface BooleanValue {
+  value: boolean;
+}
+
+/**
+ * Type definition for date value that can be handled by editor.
+ * @beta
+ */
+export interface DateValue {
+  value: Date;
+}
+
+/**
+ * Type definition for enum value that can be handled by editor.
+ * @beta
+ */
+export interface EnumValue {
+  choice: number | string;
+  label: string;
+}
+
+/**
+ *
+ */
+export type Value =
+  | NumericValue
+  | InstanceKeyValue
+  | TextValue
+  | BooleanValue
+  | DateValue
+  | EnumValue;
+
+/**
+ * Type guard for text value.
+ * @beta
+ */
+export function isTextValue(
```

Comment:
maybe it makes sense to put these type guards under `Value` namespace?

Reply:
Added namespace

Reply:
Do you think the "Value" suffix in method names are needed? IMO, `Value.isDate` > `Value.isDateValue`